packages/terra-date-picker/src/DateUtil.js

```diff
@@ -553,6 +553,36 @@ class DateUtil {
     return momentDate.format('YYYY-MM-DD');
   }
 
+  /**
+   * Increments a month by one month
+   * @param {moment} month - month to increment
+   * @return {moment} The incremented moment month
+   */
+  static incrementMonthByMonth(date, format) {
+    const momentDate = moment(date, format, true).add(1, 'months');
+    if (momentDate.year() < Number(DateUtil.MIN_YEAR)) {
+      return DateUtil.MIN_DATE;
+    } if (momentDate.year() > Number(DateUtil.MAX_YEAR)) {
+      return DateUtil.MAX_DATE;
+    }
+    return momentDate.format('YYYY-MM-DD');
+  }
+
+  /**
+   * Increments a year by one year
+   * @param {moment} year - year to increment
+   * @return {moment} The incremented moment year
+   */
+  static incrementYearByYear(date, format) {
+    const momentDate = moment(date, format, true).add(1, 'years');
+    if (momentDate.year() < Number(DateUtil.MIN_YEAR)) {
+      return DateUtil.MIN_DATE;
+    } if (momentDate.year() > Number(DateUtil.MAX_YEAR)) {
+      return DateUtil.MAX_DATE;
+    }
+    return momentDate.format('YYYY-MM-DD');
+  }
+
```

Comment:
Why do we need these functions there are already existing functions to increment/ decrement the 3 fields which are being used like this
https://github.com/cerner/terra-framework/blob/91b1c58c141b3bcd6fae3875f3442807e2322a33/packages/terra-date-picker/src/DateInput.jsx#L495
to modify dates on click of U/DOWN arrow keys

Reply:
Existing functions to increment/ decrement the 3 fields won't set current date and it will increment/decrement only on focused field and remaining fields would be empty only. So i have simplified existing +/- functionality based on focused field - https://github.com/cerner/terra-framework/pull/1999/commits/860c744021f979dc3fef08c89fc1551a75905869